src/client/scripts/cardSubmitArea.js

```diff
@@ -78,12 +82,12 @@ function findUniqueTags(tags){
     var count = 0;
 
     for(var i = 0; i < tags.length; i++){
-        if(tags[i].length !== 0){
+        if(tags[i].trim().length !== 0){
```

Comment:
As nit-picky as this is, trim() actually takes time, just do this before if()
`tags[i] = tags[i].trim();`
Now you don't have to trim for each check

Also, readability side note, nesting is the root of all evil, if you can do something with as little nesting as possible, go for it. For, with if, and another if hurts me.

Feel free to do something like 
```
85 tags[i] = tags[i].trim()
86 if(tags[i].length == 0)
87     continue;
88 if( i== 0 ){
89    // etc...
```